src/Compilers/CSharp/Portable/CodeGen/EmitAddress.cs

```diff
@@ -17,8 +17,13 @@ private enum AddressKind
             // reference may be written to
             Writeable,
 
-            // reference itself will not be written to, but may be used to modify fields.
-            ReadOnly
+            // reference itself will not be written to, but may be used for call, callvirt.
+            // for all purposes it is the same as Writeable, except when fetching an address of an array element
+            // where it results in a ".readonly" prefix to deal with array covariance.
+            Constrained,
+
+            // reference itself will not be written to, nor it will be used to modify fields.
+            Readonly,
```

Comment:
Consider reverting to `ReadOnly` rather than `Readonly`, for consistency with other enums such as `RefKind`. (It looks like we use `Readonly` in `ErrorCode`, but not elsewhere.)